app/src/main/java/com/kickstarter/ui/adapters/CommentsAdapter.kt

```diff
@@ -4,27 +4,62 @@ import android.view.LayoutInflater
 import android.view.ViewGroup
 import androidx.annotation.LayoutRes
 import com.kickstarter.R
+import com.kickstarter.databinding.CommentInitialLoadErrorLayoutBinding
 import com.kickstarter.databinding.ItemCommentCardBinding
 import com.kickstarter.ui.data.CommentCardData
 import com.kickstarter.ui.viewholders.CommentCardViewHolder
 import com.kickstarter.ui.viewholders.EmptyCommentsViewHolder
+import com.kickstarter.ui.viewholders.InitialCommentLoadErrorViewHolder
 import com.kickstarter.ui.viewholders.KSViewHolder
 
 class CommentsAdapter(private val delegate: Delegate) : KSListAdapter() {
     interface Delegate : EmptyCommentsViewHolder.Delegate, CommentCardViewHolder.Delegate
 
+    companion object {
+        private const val SECTION_INITIAL_LOAD_ERROR = 0
+        private const val SECTION_COMMENT_CARD = 1
+    }
+
+    init {
+        resetList()
+    }
+
     @LayoutRes
     override fun layout(sectionRow: SectionRow): Int {
-        return R.layout.item_comment_card
+        return when (sectionRow.section()) {
+            SECTION_COMMENT_CARD -> R.layout.item_comment_card
+            SECTION_INITIAL_LOAD_ERROR -> R.layout.comment_initial_load_error_layout
+            else -> 0
+        }
     }
 
-    fun takeData(comments: List<CommentCardData>) {
+    private fun resetList() {
         clearSections()
-        addSection(comments)
+        insertSection(SECTION_INITIAL_LOAD_ERROR, emptyList<Boolean>())
+        insertSection(SECTION_COMMENT_CARD, emptyList<CommentCardData>())
+    }
+
+    fun takeData(comments: List<CommentCardData>) {
+        resetList()
+        setSection(SECTION_COMMENT_CARD, comments)
+        submitList(items())
+    }
+
+    fun insertPageError() {
+        resetList()
+        setSection(SECTION_INITIAL_LOAD_ERROR, listOf(true))
         submitList(items())
     }
 
     override fun viewHolder(@LayoutRes layout: Int, viewGroup: ViewGroup): KSViewHolder {
-        return CommentCardViewHolder(ItemCommentCardBinding.inflate(LayoutInflater.from(viewGroup.context), viewGroup, false), delegate)
+        return when (layout) {
+            R.layout.item_comment_card -> CommentCardViewHolder(ItemCommentCardBinding.inflate(LayoutInflater.from(viewGroup.context), viewGroup, false), delegate)
+            R.layout.comment_initial_load_error_layout -> InitialCommentLoadErrorViewHolder(
```

Comment:
I think you can use delete the `InitialCommentLoadErrorViewHolder` and reuse the already existing `EmptyViewHolder` just inflate the UI you created

Reply:
yeah. done